src/ethereum_test_types/blob_types.py

```diff
@@ -1,19 +1,514 @@
 """Blob-related types for Ethereum tests."""
 
+import random
+from enum import Enum
 from hashlib import sha256
-from typing import List
+from os.path import realpath
+from pathlib import Path
+from typing import Any, ClassVar, List, Literal, cast
 
-from ethereum_test_base_types import Bytes, CamelModel, Hash
+import ckzg  # type: ignore
+import platformdirs
+
+from ethereum_test_base_types.base_types import Bytes, Hash
+from ethereum_test_base_types.pydantic import CamelModel
+from ethereum_test_forks import (
+    Cancun,
+    CancunToPragueAtTime15k,
+    Fork,
+    Osaka,
+    Prague,
+)
+
+
+def fork_string_to_object(fork_name: str) -> type[Cancun] | type[Prague] | type[Osaka]:
+    """Take a fork string and return the respective fork as object."""
+    fork_name = fork_name.lower()
+
+    if fork_name == "cancun":
+        return Cancun
+    if fork_name == "prague":
+        return Prague
+    if fork_name == "osaka":
+        return Osaka
+    raise ValueError(f"Fork {fork_name} has not yet been implemented in this function.")
+
+
+CACHED_BLOBS_DIRECTORY: Path = (
+    Path(platformdirs.user_cache_dir("ethereum-execution-spec-tests")) / "cached_blobs"
+)
 
 
 class Blob(CamelModel):
     """Class representing a full blob."""
 
     data: Bytes
-    kzg_commitment: Bytes
-    kzg_proof: Bytes | None = None
-    kzg_cell_proofs: List[Bytes] | None = None
+    commitment: Bytes
+    proof: List[Bytes] | Bytes  # Bytes < Osaka, List[Bytes] >= Osaka
+    cells: List[Bytes] | None  # None (in json: null)  < Osaka, List[Bytes] >= Osaka
+
+    versioned_hash: Hash
+    name: str  # blob_<fork>_<seed>
+    fork: str
```

Comment:
```suggestion
    fork: Fork
```
After #1686, and can also delete all `str`<->`Fork` conversions everywhere in this file.

Reply:
A side-effect of this change is that pydantic now struggles to serialize and deserialize Blob object (when we sticked to basic python types `Blob.model_validate_json(json_str)` and `self.model_dump_json()` 'just worked' but now we have to add hard-to-read logic to tell pydantic how to handle the Fork type at serialization and deserialization:

```
    @field_validator("fork", mode="before")
    @classmethod
    def validate_fork(cls, v):
        """
        When reading JSON file and trying to go back from fork string to fork object we must
        tell pydantic how to do this.
        """
        if isinstance(v, str):
            return fork_string_to_object(v)
        return v

    @field_serializer("fork")
    def serialize_fork(self, fork: Fork) -> str:
        """
        When trying to serialize a Blob object into a JSON file we must
        tell pydantic how to do this.
        """
        return fork.name()
```